dali/kernels/slice/slice_gpu.cuh

```diff
@@ -86,11 +97,15 @@ __device__ void SliceFuncNoPad(OutputType *__restrict__ out, const InputType *__
   }
 }
 
-DALI_HOST_DEV DALI_FORCEINLINE bool is_out_of_bounds(int64_t idx, int64_t data_extent) {
-  // check idx < 0 and idx >= data_extent at once
-  return static_cast<uint64_t>(idx) >= static_cast<uint64_t>(data_extent);
+template <typename T>
+DALI_HOST_DEV DALI_FORCEINLINE T ldg(const T* ptr) {
+  return __ldg(ptr);
 }
 
+template <>
+DALI_HOST_DEV DALI_FORCEINLINE bool ldg(const bool* ptr) {
+  return *ptr;
+}
```

Comment:
Is there no __ldg for bool? If so:
```suggestion
__device__ DALI_FORCEINLINE bool __ldg(const bool* ptr) {
  return __ldg(reinterpret_cast<const type_of_size<sizeof(bool)> *>(ptr));
}
```
`type_of_size` is in `kernels/common/type_erasure.h`